__tests__/unit/core-blockchain/processor/block-processor.test.ts

```diff
@@ -248,6 +248,24 @@ describe("Block processor", () => {
                 database.getActiveDelegates = getActiveDelegatesBackup;
             });
 
+            it("should reject a block with invalid nonce order", async () => {
```

Comment:
Can you instead create a new handler test like how it's done for the other handlers here: https://github.com/ArkEcosystem/core/blob/master/__tests__/unit/core-blockchain/processor/handlers/

Then it's also more clear wether it is really rejected because of the nonce order or if it's just a false positive.

Reply:
The handler is really simple:
```
export class NonceOutOfOrderHandler extends BlockHandler {
    public async execute(): Promise<BlockProcessorResult> {
        await super.execute();

        return BlockProcessorResult.Rejected;
    }
}
```

It does not make sense to write a test that creates a new class `NonceOutOfOrderHandler`, calls its `execute()` method and checks that it returns `BlockProcessorResult.Rejected` because that is already being done as part of the newly added test "should reject a block with invalid nonce order".

In `__tests__/unit/core-blockchain/processor/handlers/` we have tests only for the more complex handlers that contain their own logic.

As for the false positive, I modified the test a bit to verify that `NonceOutOfOrderHandler` will be used in 57e14f355